### PR TITLE
Add newsgroups dataset

### PR DESCRIPTION
**Pull request by RAMitchell**
For testing very sparse datasets.

### CODE REVIEW

**Comment by trivialfis** [comment]
that's a small dataset, seems to be a good fit for testing accuracy but not for performance benchmark.

**Comment by RAMitchell** [comment]
Actually it's the slowest dataset of all, taking 240 seconds even with improved xgboost split evaluation on gpu. Lightgbm cpu takes 350 seconds.
